### ISSUE TITLE
Video autoplay doesn’t work on mobile

### ISSUE THREAD

**Issue by dboulet** [8manos/async-media-load#11]
This script works well, but on mobile, autoplay for videos is not possible and users are required to tap the thumbnail twice — once to load the iframe, and again to play the video. It appears that there is no way to work around this, since platforms like iOS absolutely require user interaction with a video before it will be allowed to play.

I would suggest that on mobile, it might be a better idea to load the iframe immediately, instead of using an image thumbnail as a placeholder— but even then, there is no easy way to auto-detect support for video autoplay.

**Comment by Mantish** [comment]
You're right, that could be annoying, but the whole point of the plugin is not to load all the embeds at once.

Maybe we could have another option to choose whether to load the iframes immediately or not.

Also, I'm not sure if there's an easy way to detect if it's a mobile browser.

**Comment by dboulet** [comment]
I know, I agree that having the script immediately load all the iframes mostly defeats the purpose of the plugin. On the other hand, forcing users to tap twice on mobile leads to frustration and a bad user experience.

I like the idea of having an option for initially loading the iframes instead of image thumbnails. There would still be some speed benefit to the user with this option turned on, since the iframes could be inserted after the page has finished loading and would block page rendering. It would also allow mobile detection to be performed outside of this plugin, while still keeping the plugin fairly simple. I can file a pull request for this, but still need to determine what the name for this option should be.

**Comment by Mantish** [comment]
I guess we can call it autoload.
Thanks!!!
